core/promise/consumer.go

```diff
@@ -36,17 +62,22 @@ func (c *Consumer) NewRequest() (requestPtr interface{}) {
 
 // Consume handles requests from endpoint and replies with response
 func (c *Consumer) Consume(requestPtr interface{}) (response interface{}, err error) {
-	// request := requestPtr.(*Request)
+	request, ok := requestPtr.(*Request)
+	if !ok {
+		return failedResponse(errUnsupportedRequest), errUnsupportedRequest
+	}
 
-	// TODO there should be some validation of the received proposal and storing it somewhere for the server needs.
-	// TODO signature validation of the promise should be here too.
+	if err := request.SignedPromise.Validate(c.proposal, c.balanceRegistry); err != nil {
+		return failedResponse(err), nil
+	}
 
-	// if request.SignedPromise.IssuerSignature == valid {
-	// 		return &Response{
-	// 			Success: false,
-	// 			Message: fmt.Sprintf("Bas signature: %s", request.SignedPromise.IssuerSignature),
-	// 		}, fmt.Errorf("Bad promise signature")
-	// }
+	if err := c.storage.Store(request.SignedPromise.Promise.IssuerID, &request.SignedPromise.Promise); err != nil {
+		return nil, err
```

Comment:
`err` not returned too. We should be able to log all improper activity by client on service log...